### DISCUSSION TITLE
How to use API to update stock

### DISCUSSION THREAD

**Discussion by StormTrooper** [inventree/InvenTree#4424] [Q&A] [answered]
I'm busy trying to build some automation in that involves reading and updating stock quantities. 

Read the current stock I can do with ```/api/part/{number}``` and then pull out the ```in_stock``` value.

If I want to change the stock quantities then this is where the confusion comes in. Maybe I'm not understanding how the stock system work. 
Currently when adding a new part to the system I add "new stock item" and enter the quantities.  If I need to  reduce the stock for what ever reason (sold via another platform) I go to stock actions / count part stock and update the value.

Looking at the API document I need to use stock / update
```/api/stock/{id}/.*```
and provide id, part and quantity.  How do I know what the ```id``` should be for a particular part. 

Another query based off this is when I do find out how to use the quantity, if I want to reduce it by 2 using a API, I guess I first need to read the current stock, then subtract, and then write the new value back?

**Comment by matmair** [comment, marked as answer]
@StormTrooper you need to query the API for the stock you want - one Part can have many StockItems in many StockLocations. You could filter by part for example.

You can also call `/api/stock/remove/` or `/api/stock/add/` to decrease or increase the quantity by a set amount. Please look at the API docs for more detail https://demo.inventree.org/api-doc/#stock-add-create

**Reply by matmair** [reply]
On an administrative note: please post questions into the Q&A section

**Reply by StormTrooper** [reply]
Thanks. Will do. 

The add/remove looks like it will do the job.
